### PR TITLE
runtime: Update the chain constants.

### PR DESCRIPTION
**Pull request by shamil-gadelshin**
Changes:
- SlashDeferDuration changed to the BONDING_DURATION - 1
- Staking history_depth was increased to 336

### CODE REVIEW

File: node/src/chain_spec/mod.rs [hunk 1]
```diff
@@ -260,6 +260,7 @@ pub fn testnet_genesis(
                 .collect(),
             invulnerables: initial_authorities.iter().map(|x| x.0.clone()).collect(),
             slash_reward_fraction: Perbill::from_percent(10),
+            history_depth: 336,
```
Comment (bwhm) [inline node/src/chain_spec/mod.rs:263]: Where did 84 come from? Was it just some default value?

Comment (shamil-gadelshin) [reply node/src/chain_spec/mod.rs:263]: Yes.

**Review by bwhm — APPROVED** [review]

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -296,8 +296,8 @@ parameter_types! {
 
 parameter_types! {
     pub const SessionsPerEra: sp_staking::SessionIndex = 6;
-    pub const BondingDuration: pallet_staking::EraIndex = 24;
-    pub const SlashDeferDuration: pallet_staking::EraIndex = 6; // 1/4 the bonding duration.
+    pub const BondingDuration: pallet_staking::EraIndex = BONDING_DURATION;
+    pub const SlashDeferDuration: pallet_staking::EraIndex = BONDING_DURATION - 1; // 'slightly less' than the bonding duration.
```
Comment (mnaamani) [inline runtime/src/lib.rs:300]: If we ever (mistakenly) set BONDING_DURATION to 0 we might get problems (compiler would catch attempting to set a negative number).
, or even setting it to 1 then `SlashDeferDuration` = 0.

So maybe setting `pub const Slash_Defer_Duration = 23` then `BondingDuration` = `Slash_Defer_Duration + 1`

**Comment by mnaamani** [comment]
So I only had a small comment, but otherwise the numbers look good.

File: runtime/src/lib.rs (hunk 2 above)
Comment (shamil-gadelshin) [reply runtime/src/lib.rs:300]: The compiler prevents it with the `arithmetic operation will overflow` message.

**Review by mnaamani — APPROVED** [review]

**Pull request merged**